tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -36,6 +36,72 @@
     ASSERT_TRUE([backingElement isKindOfClass:[WXFrameworkElement class]]);
 }
 
+@interface UIDeallocTestButton : UIButton {
+    NSCondition* _condition;
+    BOOL* _signaled;
+}
+@end
+
+@implementation UIDeallocTestButton
+
+- (void)setDeallocEvent:(NSCondition*)condition signaledFlag:(BOOL*)signaled {
+    _condition = condition;
+    _signaled = signaled;
+}
+
+-(void)dealloc {
+    [_condition lock];
+    *_signaled = YES;
+    [_condition signal];
+    [_condition unlock];
+    [super dealloc];
+}
+@end
+
+TEST(UIButton, CheckForLeaks) {
+    Microsoft::WRL::WeakRef weakXamlElement;
+    {
+        __block UIButtonWithControlsViewController* buttonVC = [[UIButtonWithControlsViewController alloc] init];
+        UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+        __block UIDeallocTestButton* testButton = nil;
+        __block BOOL signaled = NO;
+        __block NSCondition* condition = [[NSCondition alloc] init];
+
+        // Create and render the button
+        dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
I'm fine waiting for this to happen before I wait.

---
In reply to: [98299832](https://github.com/Microsoft/WinObjC/pull/1861#discussion_r98299832) [](ancestors = 98299832)